j5/boards/sb/arduino.py

```diff
@@ -24,6 +25,30 @@ def __init__(
 
         self.ultrasound_sensors = UltrasoundSensors(self)
 
+    @property
+    def serial_number(self) -> str:
+        """Get the serial number."""
+        return self._serial
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """Get the firmware version of the board."""
+        return self._backend.firmware_version
+
+    @property
+    def pins(self) -> ImmutableDict[ArduinoUno.PinNumber, GPIOPin]:
+        """Get the GPIO pins."""
+        pins = ImmutableDict[ArduinoUno.PinNumber, GPIOPin]({
+            **cast(Mapping[ArduinoUno.PinNumber, GPIOPin], self._analogue_pins),
+            **cast(Mapping[ArduinoUno.PinNumber, GPIOPin], self._digital_pins),
+
+        })
+        return pins
+
+    def make_safe(self) -> None:
+        """Make this board safe."""
+        pass
```

Comment:
this looks like an unrelated change?

Reply:
Literally force-pushed just now to get rid of that 😅

Reply:
(Was an artifact of a bodged rebase)